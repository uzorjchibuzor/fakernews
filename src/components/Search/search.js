import React from 'react'
import PropTypes from 'prop-types'

const Search = ({ value, onChange, onSubmit, children }) => {
  return (
    <form onSubmit={onSubmit}>
      {children} <input
      type="text"
      value={value}
      onChange={onChange} />
      <button type="submit">
        {children}
      </button>
    </form>
  )
}

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
}

export default Search