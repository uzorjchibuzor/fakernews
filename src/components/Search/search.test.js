import React from 'react';
import ReactDOM from 'react-dom'
import Search from './search';
import renderer from 'react-test-renderer'

describe('Search', () => {

  it('renders', () => {
    const div = document.createElement('div')
    const value = 'react'
    const dummyFunc = () => 1
    ReactDOM.render(<Search value={value} onSubmit={dummyFunc} onChange={dummyFunc}>Search</Search>, div)
  })

  test('snapshots', () => {
    const value = 'react'
    const dummyFunc = () => 1

    const component = renderer.create(
      <Search value={value} onSubmit={dummyFunc} onChange={dummyFunc}>Search</Search>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    });
})