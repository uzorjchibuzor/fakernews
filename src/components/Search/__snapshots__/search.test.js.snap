// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshots 1`] = `
<form
  onSubmit={[Function]}
>
  Search
   
  <input
    onChange={[Function]}
    type="text"
    value="react"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
